src/powershell/Public/Remove-FinOpsHub.ps1

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+    .SYNOPSIS
+    Delete a FinOps Hub instance and optionally keep the storage account hosting cost data.
+
+    .PARAMETER Name
+    Required when specifying Name. Name of the FinOps Hub.
+
+    .PARAMETER ResourceGroupName
+    Optional when specifying Name. Resource Group Name for the FinOps Hub.
+
+    .PARAMETER InputObject
+    Required when specifying InputObject. Expected object is the output of Get-FinOpsHub.
+
+    .PARAMETER KeepStorageAccount
+    Optional. Indicates that the storage account associated with the FinOps Hub should be retained.
+
+    .EXAMPLE
+    Remove-FinOpsHub -Name MyHub -ResourceGroupName MyRG -KeepStorageAccount
+
+    Deletes a FinOps Hub named MyHub and deletes all associated resource except the storagea ccount.
+#>
+
+function Remove-FinOpsHub
+{
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param
+    (
+        [Parameter(Mandatory = $true, ParameterSetName = 'Name')]
+        [ValidateNotNullOrEmpty ()]
+        [string]
+        $Name,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [string]
+        $ResourceGroupName,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'Object')]
+        [ValidateNotNullOrEmpty ()]
+        [psobject]
+        $InputObject,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $KeepStorageAccount,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $Force
+    )
+
+    $context = Get-AzContext
+    if (-not $context)
+    {
+        throw $script:localizedData.ContextNotFound
+    }
+
+    try
+    {
+
+        $fhO = $null
+
+        if ($PSCmdlet.ParameterSetName -eq 'Name')
+        {
+            if (-not [string]::IsNullOrEmpty($ResourceGroupName))
+            {
+                $fhO = Get-FinOpsHub -Name $Name -ResourceGroupName $ResourceGroupName
+            }
+            else
+            {
+                $fhO = Get-FinOpsHub -Name $Name
+            }
+        }
+        else
+        {
+            $fhO = $InputObject
+        }
+
+        # TODO: $Name is not available if InputObject provided
+        if (-not $fhO)
+        {
+            throw $script:localizedData.FinOpsHubNotFound -f $Name
+        }
+
+        # Extract unique identifier from Key Vault name
+        $kv = $fhO.Resources | Where-Object ResourceType -eq "Microsoft.KeyVault/vaults"
```

Comment:
We may need to account for soft deletes of key vaults. @flanakin do we want to support purging key vaults?

Reply:
Don't care right now, as I have no idea what to delete and trying to find a way forward... but yes, in ALZ, this will not work as soft-delete is delete protection is enabled by default.